sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -668,6 +669,573 @@ protected virtual BlobBaseClient WithSnapshotImpl(string snapshot)
         }
         #endregion Download
 
+        #region Parallel Download
```

Comment:
Do we need both Download and Parallel Download?  Wouldn't Parallel Download suffice for everything?

Reply:
My view was that parallel depended on sequential.  There's certainly a limited below which sequential is the way to go.  Parallel is primarily a large-file thing, both in terms of how it handles memory and in having an extra call to detect the size and determine whether to use the non-parallel approach.

Another point is that parallel download depends on being handed the destination, rather than being returned as part of BlobDownloadInfo (hence why I'm only returning BlobProperties).  They have different signatures because of that.